File: src/main/java/com/hodvidar/insuratradeflow/mvc/api/dto/InsurancePolicyDto.java
package com.hodvidar.insuratradeflow.mvc.api.dto;

import java.time.LocalDate;
import java.time.LocalDateTime;

public record InsurancePolicyDto(
        Long id,
        String name,
        String status, // ACTIVE or INACTIVE
        LocalDate startDate,
        LocalDate endDate,
        LocalDateTime creationDateTime,
        LocalDateTime updateDateTime
) {
}

File: backend/src/node/api/dtos/insurance-policy.dto.ts
export class InsurancePolicyDto {
    id?: number;
    name: string;
    status: string;  // enum as string
    startDate: string;  // ISO date format (YYYY-MM-DD)
    endDate: string;    // ISO date format (YYYY-MM-DD)
    creationDateTime?: string;  // ISO timestamp
    updateDateTime?: string;    // ISO timestamp
}